test/TestRist.cpp

```diff
@@ -0,0 +1,329 @@
+        std::fill(sendBuffer.begin(), sendBuffer.end(), i + '0');
+        EXPECT_TRUE(sender.sendData((const uint8_t*)sendBuffer.data(), sendBuffer.size()));
+    }
+
+    {
+        std::unique_lock<std::mutex> lock(receiverMutex);
+        bool successfulWait = receiverCondition.wait_for(lock, std::chrono::seconds(2),
+                                                         [&]() { return nReceivedPackets == kSentPackets; });
+        EXPECT_TRUE(successfulWait) << "Timeout waiting for receiving data from sender, sent " << kSentPackets
+                                    << " packet(s) and received " << nReceivedPackets << " packet(s).";
+    }
+}
+
+// TODO Enable test when STAR-38 is fixed.
+TEST(TestRist, DISABLED_TestPsk) {
+    RISTNetReceiver receiver;
+    std::vector<std::string> receiverInterfaces{"rist://@0.0.0.0:8000"};
+    RISTNetReceiver::RISTNetReceiverSettings receiverSettings;
+    receiverSettings.mPSK = kValidPsk;
+
+    receiver.validateConnectionCallback = [&](const std::string& ipAddress, uint16_t port) {
+        return std::make_shared<RISTNetReceiver::NetworkConnection>();
+    };
+
+    ASSERT_TRUE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+
+    std::vector<std::tuple<std::string, int>> senderInterfaces{
+        std::tuple<std::string, int>("rist://127.0.0.1:8000", 0)};
+    RISTNetSender::RISTNetSenderSettings senderSettings;
+    senderSettings.mPSK = kInvalidPsk;
+    RISTNetSender sender;
+    EXPECT_FALSE(sender.initSender(senderInterfaces, senderSettings));
+
+    senderSettings.mPSK = kValidPsk;
+    EXPECT_TRUE(sender.initSender(senderInterfaces, senderSettings));
+}
+
+TEST(TestRist, LargeMessage) {
+    RISTNetReceiver receiver;
+    const uint16_t kBufferSize = 10'000 - 32; // Largest packet allowed: RIST_MAX_PACKET_SIZE - 32
+    std::vector<std::string> receiverInterfaces{"rist://@0.0.0.0:8000"};
+    RISTNetReceiver::RISTNetReceiverSettings receiverSettings;
+    rist_peer_config_defaults_set(&receiverSettings.mPeerConfig);
+    receiverSettings.mPSK = kValidPsk;
+    auto receiverCtx = std::make_shared<RISTNetReceiver::NetworkConnection>();
+
+    receiver.validateConnectionCallback = [&](const std::string& ipAddress, uint16_t port) { return receiverCtx; };
+    ASSERT_TRUE(receiver.initReceiver(receiverInterfaces, receiverSettings));
+
+    std::vector<std::tuple<std::string, int>> senderInterfaces{
+        std::tuple<std::string, int>("rist://127.0.0.1:8000", 0)};
+
+    RISTNetSender::RISTNetSenderSettings senderSettings;
+    senderSettings.mPSK = kValidPsk;
+    RISTNetSender sender;
+    ASSERT_TRUE(sender.initSender(senderInterfaces, senderSettings));
+
+    std::vector<uint8_t> sendBuffer(kBufferSize);
+    std::fill(sendBuffer.begin(), sendBuffer.end(), 1);
+    EXPECT_TRUE(sender.sendData((const uint8_t*)sendBuffer.data(), sendBuffer.size()));
+
+    std::vector<uint8_t> largeBuffer(kBufferSize + 1);
+    std::fill(largeBuffer.begin(), largeBuffer.end(), 1);
```

Comment:
There is a constructor for this (fill constructor), just add 1 as an argument on the line above (and don't use curly braces!)